### DISCUSSION TITLE
Riverpod generator - Provider function params must have a consistent hashCode and ==?

### DISCUSSION THREAD

**Discussion by benji-farquhar** [rrousselGit/riverpod#1928] [Q&A] [answered]
The Riverpod docs on family stress that:

>For families to work correctly, it is critical for the parameter passed to a provider to have a consistent hashCode and ==.

The Riverpod generator docs say that:

> Passing parameters to providers is now unrestricted. Instead of being limited to using [family](https://docs-v2.riverpod.dev/docs/concepts/modifiers/family) and passing a single positional parameter, you can now pass any form of parameter. This includes named parameters, optional ones, and even default values.

But I can see that the generated code has made a `Family<YourCalss>`.

I see it generates overrides for `hashCode` and `==`, but does this still mean we should only pass parameters to the `@riverpod ` provider function, that have a consistent hashCode and ==?

**Comment by rrousselGit** [comment, marked as answer]
Yes.
They don't necesserily need to override ==/hashCode, but the == of parameters should be consistent across widget rebuild.
